packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -97,33 +96,33 @@ export class ApplicationExtension<
     }
 
     /**
-     * Called on the server to serialize the extension data that will be sent to the client.
+     * Called on the server to serialize the extension's routes that will be sent to the client.
      *
-     * @returns SerializedExtension - The serialized extension data.
+     * @returns SerializedRouteProps - The serialized extension data.
      * @returns null if there's no asynchronous state that needs to be serialized
      * @throws Error if the routes have not been loaded.
      */
-    public serialize(): SerializedExtension | null {
+    public serializeRoutes(): SerializedRouteProps[] | null {
```

Comment:
since a serialized extension only contains routes i've renamed this method to just `serializeRoutes`.
In the future if we serialize more data we can make a `serialize` method that calls this method

Reply:
Yeah, I'm ok with having both those methods in the same file (ApplicationExtension).

Aside: because of this [comment thread](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2342/files#r2021294123), I was thinking we actually do need the component map.. and similarly for serialize and deserialize. Anyways, I'm saying that I'm now convinced we need all of them, and I like the direction this PR takes.